ReactAndroid/src/main/java/com/facebook/react/modules/location/LocationModule.java

```diff
@@ -188,18 +190,34 @@ public void stopObserving() {
     mWatchedProvider = null;
   }
 
+
+  private boolean isProviderEnabled(LocationManager locationManager, String provider) {
+    try {
+      // Before API version LOLLIPOP, this method would throw SecurityException
+      // if the location permissions were not sufficient to use the specified provider.
+      return locationManager.isProviderEnabled(provider);
+    } catch (SecurityException e) {
+      return false;
+    }
+  }
+
   @Nullable
-  private static String getValidProvider(LocationManager locationManager, boolean highAccuracy) {
+  private String getValidProvider(LocationManager locationManager, boolean highAccuracy) {
     String provider =
         highAccuracy ? LocationManager.GPS_PROVIDER : LocationManager.NETWORK_PROVIDER;
-    if (!locationManager.isProviderEnabled(provider)) {
+    if (!isProviderEnabled(locationManager, provider)) {
       provider = provider.equals(LocationManager.GPS_PROVIDER)
           ? LocationManager.NETWORK_PROVIDER
           : LocationManager.GPS_PROVIDER;
-      if (!locationManager.isProviderEnabled(provider)) {
+      if (!isProviderEnabled(locationManager, provider)) {
         return null;
       }
     }
+    // If it's an enabled provider, but we don't have permissions, ignore it
+    int finePermission = ContextCompat.checkSelfPermission(getReactApplicationContext(), android.Manifest.permission.ACCESS_FINE_LOCATION);
```

Comment:
To reply to your comment directly (as opposed to the main thread, where it appears to have been lost):

Default RN apps request FINE permissions, and so RN apps declaring FINE do not have any issues. My app only declares COARSE, and so wanted to fix RN for my use case.

I am calling `navigator.geolocation.getCurrentPosition(..., {enableHighAccuracy: false})` to avoid using the GPS and just rely on the NETWORK provider.

To your question: yes, by and large the Geolocation library works just fine. Unfortunately, if there is a problem with the NETWORK provider [not being enabled](https://stackoverflow.com/questions/37440189/locationmanager-gives-network-provider-disabled), then the RN code attempts to fall back to using the GPS provider.

And if this GPS provider is enabled, my app will crash with an unhandled Java exception, because it only declares COARSE permission. I am merely checking for FINE permission before using the GPS provider, to avoid this crash.